### ISSUE TITLE
 Error getting token form google

### ISSUE THREAD

**Issue by pieotr** [1nikolas/play-integrity-checker-app#17]
I've just got an error message, error getting token form google
The calling app is making too many requests to the API and hance is throttled.
This shouldn't happen. If it does please open an issue on Github.

i hav used this app to chck my intgeity status, becuse play certification checking app say it passes but play store and gpay say my phone is not certified. I have Xaomi mi 9 and i was using EvolutionX rom and i had root instaled. It was all working fine but after a while form one day to another it stoped and while trying to pay said my device is not meeting security requirements. Done factory reset already, clean flush of ROM, other roms as well, and it's still not woroking even thou i havent installed root yet, just basic rom that sould have all certifications. Can someone help me?

**Comment by 1nikolas** [comment]
Yes y'all hit the 10k/day request limit. I'll try to request an increase

**Comment by nutterthanos** [comment]
I wonder how that happened?

**Comment by 1nikolas** [comment]
> I wonder how that happened?

It's simple. Too many people used the app today and Google allows certain uses (aka requests) of the api per day. You can tequest to increase it if you want (that's what I did) but it's up to them if they do it. The limit resets midnight (PT) which is in ~7 hours.

**Comment by Jon8RFC** [comment]
Would an alternate solution be multiple tokens, which the app cycles through?

It needs to be legitimate, but I'm happy to be one of the people to make a legitimate API token registration and "donate" it to the cause, assuming it's free. I'll register my email and phone number or whatever is necessary.

**Comment by stoag** [comment]
Possibly add a setting to use our own API key? 

Having the same issue and my NFC just stopped working. Didn't find out till I was at checkout and didn't have my wallet. Need to troubleshoot asap. This app would be really helpful.

**Comment by da411d** [comment]
This is mass problem, because a lot of people checking their safetynet, because Google Pay stopped working on most of alternative ROMs🥲

**Comment by 1nikolas** [comment]
I have no idea why Github doesn't send me emails about replies here. Google has responded about the api increase:

> Our internal teams are currently reviewing your Play Integrity API quota increase request.

**Comment by 1nikolas** [comment]
> Would an alternate solution be multiple tokens, which the app cycles through?
> 
> It needs to be legitimate, but I'm happy to be one of the people to make a legitimate API token registration and "donate" it to the cause, assuming it's free. I'll register my email and phone number or whatever is necessary. 

We can do it officially (if google wants to increase my api limits). Using multiple api keys to bypass quota limits I think is against TOS

**Comment by 1nikolas** [comment]
Also now that I think about it, we cant do any of that. The google cloud project (which has the api key) must be linked to play console and you can only link one per app.

**Comment by mskonovalov** [comment]
Same here, apparently Google releases new version of integrity check, a lot of people got Gpay rejecting to work and started verifying the integrity 🤷

**Comment by dongltron** [comment]
Yup, that's what's happening.

**Comment by Iwearthebelt** [comment]
Any news or updates on this issue? Been looking around since yesterday but not having any luck. Are there any threads and discussions about this anywhere? Thanks

**Comment by mskonovalov** [comment]
> Any news or updates on this issue? Been looking around since yesterday but not having any luck. Are there any threads and discussions about this anywhere? Thanks

What exactly you are looking for?
There is an alternative app on Google Play, just search "integrity check" or something similar.

Or you a re looking for a fix to overcome new change from the Google?
If latter then try https://github.com/chiteroman/PlayIntegrityFix

**Comment by 1nikolas** [comment]
Unfortunately can't do anything. Google doesn't like my app :(

> We’re unable to fulfill your request for Consumer Project ID ... for the requested 10,000 to 1,000,000 (10K to 1M) queries per day as we have identified that your method of validating Play Integrity API responses is incorrect.

**Comment by 1nikolas** [comment]
You can try this great app someone made it's way better than mine

https://play.google.com/store/apps/details?id=com.henrikherzig.playintegritychecker

**Comment by 1nikolas** [comment]
Asked for clarification and this is what the told me:

> Upon checking with our team, they were able to identify that your app offers the same functionality in one of our existing tools which you may use instead of requesting a quota increase.

Aka since my app does exactly what Play Store does (via developer options) they can't give me more quota.

**Comment by Unlearned6688** [comment]
Thank you for the app and the explanation

Giant corporations gonna giant corporation... Sucks...

**Comment by thesweetiger** [comment]
> Asked for clarification and this is what the told me:
> 
> > Upon checking with our team, they were able to identify that your app offers the same functionality in one of our existing tools which you may use instead of requesting a quota increase.
> 
> Aka since my app does exactly what Play Store does (via developer options) they can't give me more quota.

this is an illegal reason , europe ,usa,canda antitrust law

**Comment by 1nikolas** [comment]
> > Asked for clarification and this is what the told me:
> > > Upon checking with our team, they were able to identify that your app offers the same functionality in one of our existing tools which you may use instead of requesting a quota increase.
> > 
> > 
> > Aka since my app does exactly what Play Store does (via developer options) they can't give me more quota.
> 
> this is an illegal reason , europe ,usa,canda antitrust law

And what are you going to do? Fight Google on court?

**Comment by SpeedyCraftah** [comment]
I guess you could somewhat bypass this by having 2 of the same app on the play store, and if one can't do anymore attestations it will redirect you to the second app and vice versa which basically means you get 20k instead, although it's not really fair for you to spend money uploading the app, if Google will even approve it that is 😂.

Hard to say whether this is another "anti-root" thing since there are other ways to verify your attestation, but it's definitely annoying.

**Comment by sukabai** [comment]
![Screenshot_2024-10-27-00-38-25-362_gr nikolasspyr integritycheck-edit](https://github.com/user-attachments/assets/32ca0ab4-ce3d-4be6-a668-f2f978278c9e)

**Comment by 1nikolas** [comment]
@sukabai does this happen every time?

**Comment by sukabai** [comment]
Yes,  started 3 days ago. Tried in the morning, noon, evening, same error.

**Comment by 1nikolas** [comment]
> Yes,  started 3 days ago. Tried in the morning, noon, evening, same error. 
> This is from Spic and Google play store:
> ![Screenshot_2024-10-27-07-19-13-574_com android vending-edit](https://github.com/user-attachments/assets/77fa6cd0-8449-4e10-b1fe-c6361db6a97a)
> ![Screenshot_2024-10-27-00-30-12-215_com henrikherzig playintegritychecker-edit](https://github.com/user-attachments/assets/b6f52581-5dd6-408c-9c4a-ad5f754e0756)
> 

Looks like a problem with your device i guess.. No idea

**Comment by sukabai** [comment]
> @sukabai does this happen every time?

Yes, even now. I think it's my phone not your app. Somehow I did something wrong and the result is that error. The problem started a few days ago, if I remember correctly, after I installed XPL-EX and played a little with it.

**Comment by sukabai** [comment]
Enabling store in Hide my applist was culprit!
